packages/strapi-plugin-content-manager/config/functions/bootstrap.js

```diff
@@ -15,17 +15,32 @@ module.exports = async cb => {
     'primaryKey',
     'associations'
   ]);
-  
+
+  const pluginsLayout = Object.keys(strapi.plugins).reduce((acc, current) => {
```

Comment:
Please split the file, it's too big and complicated to read. Also, please add comments to be able to understand each part of the code.